lablup/backend.ai-webui
# feat: connect ssh / sftp in filebrowser dialog

- Related PR: https://github.com/lablup/backend.ai/pull/1238
- How to test
  - pull `cr.backend.ai/testing/filebrowser:23.04-ubuntu20.04` image (or `cr.backend.ai/testing/filebrowser:23.04-ubuntu20.04-arm64`) and rescan
  - Etcd setting: `./backend.ai mgr etcd put volumes/proxies/<volume>/sftp_scaling_groups "ssh"`.
    - For me, I set `./backend.ai mgr etcd put volumes/proxies/local/sftp_scaling_groups "ssh"`
    - ref: https://github.com/lablup/backend.ai/pull/1246)
  - Check the DB is up-to-date.
  - Check the resources are enough. It needs (CPU: 1, MEM: 256m). If not, adjust [these lines](https://github.com/lablup/backend.ai-webui/pull/1694/files#diff-b810064a5174ba45c2487b7e92a6aaacd7349f7f637602afd38ad22bc49ba9b0R2841-R2842) for testing.

## Screenshots
- Add SSH / SFTP button: this button is active when the system role image exists.
<img width="617" alt="image" src="https://user-images.githubusercontent.com/28584164/234202707-feee3b91-e6b4-451a-9aec-ad60d4c7187e.png">

- Progress
<img width="287" alt="image" src="https://user-images.githubusercontent.com/28584164/234203004-6a19430a-8dab-4b96-93c0-5bf88d55350d.png">

- After indicator is 100%, SSH connection dialog appears.
  - you can download the ssh private key by clicking the ssh key download button.
<img width="358" alt="image" src="https://user-images.githubusercontent.com/28584164/234203203-c5473244-0d52-44e3-83d2-9dd9c85493bf.png">

- Filebrowser image or `systemSSHImage` (it is set from the `config.toml`) session is created. And this session doesn't affect resource occupancy.
<img width="1228" alt="image" src="https://user-images.githubusercontent.com/28584164/234203584-55940aa4-c630-41a9-9248-7bceeef4d0f4.png">
